activerecord/test/cases/time_precision_test.rb

```diff
@@ -0,0 +1,65 @@
+require 'cases/helper'
+
+if current_adapter?(:PostgreSQLAdapter) || mysql_56?
```

Comment:
This pattern is appearing a lot. How about we add a `supports_datetime_with_precision?` to the connection adapter, and use that?